docs/tools/flamegraphs.md

```diff
@@ -0,0 +1,167 @@
+
+To change the period of capture adjust the sleep duration (which is in seconds).
+
+The `pgrep` command is used to find the process id to capture against. Feel free to pass a number to the `-p` flag if you know the process id, or adjust the file path if the executable is in a different location.
+
+```sh
+admin@12.34.56.78: sudo perf record -F 99 -p $(pgrep -f '/usr/src/lodestar/packages/cli/bin/lodestar beacon') -g -- sleep 60
+admin@12.34.56.78: sudo perf script -f > perf.out
+admin@12.34.56.78: sudo chmod 777 ~/beacon/perf.out
+```
+
+And then copy the `perf.out` file to your local machine to render the flamegraph. Running at `99Hz` for 180 seconds results in a file size of about 3.5mb and `997Hz` for 60 seconds is roughly 4.4mb.
+
+```sh
+scp admin@12.34.56.78:/home/devops/beacon/out.perf /some_temp_dir/perf.out
+```
+
+## Rendering a Flamegraph
+
+By far the best tool to render flamegraphs is [`flamescope`](https://github.com/Netflix/flamescope) from Netflix.  It allows for easy analysis and zooming into specific time periods.  It also give a holistic view of how the process is performing over time.
+
+## Installation
+
+Python3 is required. Clone the repository and install the dependencies:
+
+```sh
+# The original is no longer maintained and had a configuration bug
+# This is a fork that fixes the issue.
+git clone https://github.com/matthewkeil/flamescope
+cd flamescope
+pip3 install -r requirements.txt
+yarn
+```
+
+## Usage
+
+```sh
+mv /some_temp_dir/perf.out /path/to/flamescope/examples
+yarn dev
+```
+
+Then navigate in a browser to `http://localhost:8080` and begin analyzing the data.
+
+<img src="../images/flamescope/home-screen.png" alt="flamescope home screen" width="1024"/>
+<img src="../images/flamescope/time-series-view.png" alt="flamescope home screen" width="1024"/>
+<img src="../images/flamescope/selecting-series.png" alt="flamescope home screen" width="1024"/>
+<img src="../images/flamescope/unfiltered-flamegraph.png" alt="flamescope home screen" width="1024"/>
+<img src="../images/flamescope/zoom-in.png" alt="flamescope home screen" width="1024"/>
+
+## Filtering Results
+
+There can be a lot of "noise" in the stack traces with `libc`, `v8` and `libuv` calls.  It is possible to filter the results to make it more useful, but note this will skew the results. Looking at the graph both filtered and unfiltered can be beneficial. The following `sed` command will remove the noise from the stack traces.  You can also use the bottom script programmatically to add/remove filters.
+
+```sh
+sed -r -e "/( __libc_start| uv_| LazyCompile | v8::internal::| node::| Builtins_| Builtin:| Stub:| LoadIC:| \\[unknown\\]| LoadPolymorphicIC:)/d" -e 's/ LazyCompile:[*~]?/ /'
+```
```

Comment:
With the one-liner sed command the ts script below should not be necessary

Reply:
Updated.